presto-verifier/src/main/java/com/facebook/presto/verifier/framework/LimitQueryDeterminismAnalyzer.java

```diff
@@ -262,7 +268,7 @@ private LimitQueryDeterminismAnalysis analyzeLimitNoOrderBy(Query newLimitQuery,
         if (rowCountHigherLimit == rowCount) {
             return DETERMINISTIC;
         }
-        if (rowCount >= limit && rowCountHigherLimit > rowCount) {
```

Comment:
why are we removing```rowCount >= limit``` check? It makes sense to be non-deterministic in this case as well.

Reply:
The change we're adding in this PR will short circuit cases with `rowCount < limit`. Therefore, when reaching this line, it is guaranteed that `rowCount >= limit`.